co-noir/ultrahonk/src/parse/types.rs

```diff
@@ -0,0 +1,1259 @@
+use super::builder::{GenericUltraCircuitBuilder, UltraCircuitBuilder, UltraCircuitVariable};
+use super::plookup::{BasicTableId, MultiTableId};
+use crate::decider::polynomial::Polynomial;
+use crate::types::ProvingKey;
+use crate::Utils;
+use ark_ec::pairing::Pairing;
+use ark_ff::{One, PrimeField, Zero};
+use itertools::izip;
+use num_bigint::BigUint;
+use std::array;
+use std::cmp::Ordering;
+use std::collections::{HashMap, HashSet};
+use std::hash::{Hash, Hasher};
+use std::ops::{Index, IndexMut};
+
+#[derive(Default, PartialEq, Eq)]
+pub(crate) struct PolyTriple<F: PrimeField> {
```

Comment:
these types and the crate in general could use more documentation

Reply:
Let's make a docu pass after the merge